src/workflow-handler.ts

```diff
@@ -127,64 +127,48 @@ export class WorkflowHandler {
       return this.workflowRunId;
     }
     try {
-      core.debug('Get workflow run id');
+      const workflowId = await this.getWorkflowId();
+
+      const response = await this.octokit.rest.actions.listWorkflowRuns({
+        owner: this.owner,
+        repo: this.repo,
+        workflow_id: workflowId,
+        event: 'workflow_dispatch',
+        branch: this.ref,
```

Comment:
@LudovicTOURMAN nice work. It makes sense to combine these 2 methods, as they use this same function.

For the `branch` argument, using `this.ref` has issues. If the user provides a simple `branch` name, it will work fine. However, the default value is `github.context.ref` which includes `refs/heads/` and will cause `workflow_runs` to return as empty. The provided `ref` could also be a `tag` or a `commit sha` which will also return empty. To guarantee a `branch` is passed to the function, we need additional steps using the `ref` (potentially with `octokit.rest.git.getRef`).

To include a test for this, you can use what I had in my closed PR: [.github/workflows/build-test.yaml](https://github.com/aurelien-baudet/workflow-dispatch/pull/15/files#diff-bd4e577d8cce18da4cecc900fca127f9a0d57d706c9cdb3507b778c1d574a54f). Note: GitHub has a weird bug (or feature?) where the workflows of a forked repo will not show and are not callable until they've run at least once (in your fork). So, the `build-test` will run if you add a `pull_request` trigger and the `echo-3` workflow is callable if you add it. The other existing workflows being tested will not be found.

Reply:
Thanks for the feedback @evanlindsey.

I looked to improve the ref/branch filter, but the benefit from that vs complexity seems not worth it.
Something that could be improved in another PR if people are having heavy loads and major response time.
In the context of that PR, it sounds easier to revert that change.

Thanks also for the testing workflow, it allows me to introduce a new test.
I'm still having trouble making them work on my fork (even after following your tips regarding the bug).
Maybe something related to permissions. My token properly has `repo` and `workflow` permission but I might be missing something else :thinking:

Reply:
Sounds good! Since the existing call to `listWorkflowRuns` doesn't include `branch` then that behavior isn't being altered... Glad to hear the tests helped out. I see that you were able to trigger all of the workflows. Nice addition with checking the named runs in parallel!